src/pyroboplan/visualization/set_get_joints.py

```diff
@@ -0,0 +1,98 @@
+from pyroboplan.core.utils import (
+    check_collisions_at_state,
+    check_valid_pose,
+    check_within_limits,
+)
+
+
+class NamedJointConfigurationsOptions:
+    """Options for named joint configurations"""
+
+    def __init__(self, allow_collisions=False):
+        """
+        Initializes a set of named joint configuration options.
+
+        Parameters
+        ----------
+            allow_collisions : bool
+                Whether to allow in-collision states to be stored.
+        """
+        self.allow_collisions = allow_collisions
+
+
+class NamedJointConfigurations:
+    """Named joint configurations helper
+
+    This is a simple helper to get, set, and visualize joint states for a given model.
+    Each model should have its own NamedJointConfigurations, and validation is performed to
+    ensure joint states are dimensionally valid and, optionally, within joint limits.
+
+    """
+
+    def __init__(
+        self, model, collision_model, options=NamedJointConfigurationsOptions()
+    ):
+        """
+        Creates an instance of a named joint configurations object.
+
+        Parameters
+        ----------
+            model : `pinocchio.Model`
+                The model to use for this named joint configurations.
+            collision_model : `pinocchio.Model`
+                The model to use for collision checking.
+            options : `NamedJointConfigurationsOptions`, optional
+                The options to use for this named joint configurations object.
+                  If not specified, default options are used.
+        """
+        self.model = model
+        self.collision_model = collision_model
+        self.data = self.model.createData()
+        self.collision_data = self.collision_model.createData()
+        self.options = options
+        self.configuration_dict = {}
+
+    def __setitem__(self, key, state):
+        """
+        Set the configuration at key to state
+        """
+        # first, check if the state is valid
+        if not check_valid_pose(self.model, state):
+            if not check_within_limits(self.model, state):
+                raise ValueError("Tried to add state outside of model's joint limits")
+            raise ValueError(
+                f"Tried to add state of dimension {len(state)} to model with number of joints {self.model.nq}"
+            )
+
+        # then, check collisions if necessary
+        if not self.options.allow_collisions:
+            if check_collisions_at_state(self.model, self.collision_model, state):
```

Comment:
you can just `and` these 2 instead of more nesting -- but this is exactly the "check and early return" pattern you should use for the previous checks too